### PR TITLE
Non root tests

### PR DESCRIPTION
**Pull request by jodh-intel**

### CODE REVIEW

**Comment by jodh-intel** [comment]
Travis seems to be borked atm.

File: virtcontainers_test.go [hunk 1]
```diff
@@ -102,6 +103,7 @@ func TestMain(m *testing.M) {
 	podDirLock = filepath.Join(runStoragePath, testPodID)
 	podFileState = filepath.Join(runStoragePath, testPodID, stateFile)
 	podFileLock = filepath.Join(runStoragePath, testPodID, lockFileName)
+	testBinHookPath = filepath.Join(testDir, "bin", "hook")
```
Comment (sboeuf) [inline virtcontainers_test.go]: Unfortunately, this won't work. Look at travis.yml, we are building and copying the hook binary at that moment, meaning that /tmp/bin/hook is a hardcoded value. The dynamic directory created by ioutil.TempDir() won't fit.

File: virtcontainers_test.go [hunk 2]
```diff
@@ -31,6 +31,8 @@ const testImage = "image"
 const testHypervisor = "hypervisor"
 const testBundle = "bundle"
 
+const testDisabledAsNonRoot = "Test disabled as requires root privs"
```
Comment (sboeuf) [inline virtcontainers_test.go]: "privileges" instead of "privs" would be better, wouldn't it ?

File: virtcontainers_test.go (hunk 1 above)
Comment (jodh-intel) [inline virtcontainers_test.go]: yep - I just discovered the interesting travis config when I raised the PR.

File: virtcontainers_test.go (hunk 2 above)
Comment (jodh-intel) [inline virtcontainers_test.go]: Agreed.

File: pkg/cni/cni_test.go [hunk 3]
```diff
@@ -391,9 +391,21 @@ func TestMain(m *testing.M) {
 		os.Exit(1)
 	}
 
+	testNetNsPath, err = ioutil.TempDir("", "virtcontainers-test-ns-")
```
Comment (sboeuf) [inline pkg/cni/cni_test.go]: I think this won't work. The testNetNsPath will be overriden by testNetNsPath assignments in other test functions, won't it ?

**Comment by jodh-intel** [comment]
Aargh - it looks like my adding a `t.Skip()` to `TestStartStopContainerHyperstartAgentSuccessful()` has pushed it over the cyclomatic complexity threshold. Any ideas (aside from bumping the threshold by 1 (that test has value `16` and the threshold is `15`))?

**Comment by sboeuf** [comment]
@jodh-intel The only clean solution is to factorize some code inside a new function...

**Comment by jodh-intel** [comment]
@sameo, @sboeuf - I think we can remove the `do-not-merge` label now - I don't have privs to do that (sniff :cry: )

**Comment by sboeuf** [comment]
@sameo if you are okay with this PR, I can merge it.

**Comment by sameo** [comment]
LGTM

**Pull request merged**